### PR TITLE
contrib/gorm.io/gorm.v1: implement gorm.io Plugin interface

### PR DESCRIPTION
**Pull request by Abyfall**
### What does this PR do?

Makes the `contrib/gorm.io/gorm.v1` package implement the gorm.io Plugin interface.

### Motivation

Easier integration with project using gorm

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by Abyfall** [comment]
[[sc-10098](https://app.shortcut.com/adagio-dot-io/story/10098/trace-gorm-queries)]

File: contrib/gorm.io/gorm.v1/gorm.go [hunk 1]
```diff
@@ -33,6 +33,21 @@ const (
 	gormSpanStartTimeKey = key("dd-trace-go:span")
 )
 
+type GormTracePlugin struct{}
```
Comment (System-Glitch) [inline contrib/gorm.io/gorm.v1/gorm.go]: Can you add options as struct field so they can be passed to `withCallbacks`?

**Review by System-Glitch — CHANGES_REQUESTED** [review]

**Comment by System-Glitch** [comment]
Please open this PR on the official repository as well when it's ready. Maybe it can be merged 🤞🏻

File: contrib/gorm.io/gorm.v1/gorm.go (hunk 1 above)
Comment (Abyfall) [reply contrib/gorm.io/gorm.v1/gorm.go]: Done

**Review by System-Glitch — APPROVED** [review]

**Pull request merged**